File: src/main/java/com/IconPippi/simtriggers/triggers/ConnectionOpenTrigger.java
package com.IconPippi.simtriggers.triggers;

public class ConnectionOpenTrigger extends Trigger {
	
	protected ConnectionOpenTrigger(String method) {
		super(method);
	}

	@Override
	protected void trigger() {
		this.getScriptLoader().invokeFunction(method, (Object[]) null);
	}

}

File: src/main/kotlin/com/simtriggers/fsx/triggers/ConnectionOpenTrigger.kt
package com.simtriggers.fsx.triggers

/**
 * 05/08/2019
 * This class represents a connection open trigger
 *
 * @author IconPippi
 */
class ConnectionOpenTrigger(private val function: String) : Trigger(function, null) {

    /**
     * Run the trigger
     */
    override fun trigger(eventName: String?) {
        scriptLoader.invokeFunction(function, null)
    }

}